nuxeo-mongodb-directory/src/main/java/org/nuxeo/directory/mongodb/MongoDBDirectory.java

```diff
@@ -93,14 +116,44 @@ public Session getSession() throws DirectoryException {
             session.getCollection(countersCollectionName).insertOne(MongoDBSerializationHelper.fieldMapToBson(seq));
         }
 
-        if (!initialized && descriptor.getDataFileName() != null && !session.hasCollection(getName())) {
-            DirectoryCSVLoader.loadData(descriptor.getDataFileName(), descriptor.getDataFileCharacterSeparator(),
-                    schema, session::createEntry);
+        if (!initialized) {
+            String policy = descriptor.getCreateTablePolicy();
+            MongoCollection collection = session.getCollection(getName());
+            switch (policy) {
+            case CREATE_TABLE_POLICY_ALWAYS:
+                collection.drop();
+                loadData(schema, session);
+                break;
+            case CREATE_TABLE_POLICY_ON_MISSING_COLUMNS:
+                if (session.hasCollection(getName())) {
+                    boolean missingColumns = schema.getFields().stream().map(f -> f.getName().getLocalName()).anyMatch(
+                            fname -> collection.count(Filters.exists(fname, false)) > 0);
```

Comment:
I'd rather be conservative and consider that a column is missing if none of the documents contain a value for the field.